syntax = "proto3";

package tiramisuserver;

// The string reversal service definition.
service TiramisuServer {
    // Function invoked to send the request
    rpc getLegality (TiramisuProgram) returns (TiramisuResult) {}
}

enum Operation {
    LEGALITY = 0;
    EXECUTION = 1;
}

// The request message containing the original string
message TiramisuProgram {
    string name = 1;
    string schedule = 2;
    Operation operation = 3;

}

// The response message containing the reversed string
message TiramisuResult {
    string name = 1;
    bool legality = 2;
    string isl_ast = 3;
    string execution_times = 4;
}
